virtcontainers/fc.go

```diff
@@ -698,22 +663,64 @@ func (fc *firecracker) startSandbox(timeout int) error {
 		}
 	}
 
-	fc.fcSetVMRootfs(image)
+	if err := fc.fcSetVMRootfs(image); err != nil {
+		return err
+	}
+
 	if err := fc.createDiskPool(); err != nil {
 		return err
 	}
 
+	fc.state.set(cfReady)
 	for _, d := range fc.pendingDevices {
-		if err = fc.addDevice(d.dev, d.devType); err != nil {
+		if err := fc.addDevice(d.dev, d.devType); err != nil {
 			return err
 		}
 	}
 
-	if err := fc.fcStartVM(); err != nil {
+	return nil
+}
+
+// startSandbox will start the hypervisor for the given sandbox.
+// In the context of firecracker, this will start the hypervisor,
+// for configuration, but not yet start the actual virtual machine
+func (fc *firecracker) startSandbox(timeout int) error {
+	span, _ := fc.trace("startSandbox")
+	defer span.Finish()
+
+	if err := fc.fcInitConfiguration(); err != nil {
 		return err
 	}
 
-	return fc.waitVMM(timeout)
+	data, errJSON := json.MarshalIndent(fc.fcConfig, "", "\t")
```

Comment:
`MarshalIndent` why not just `Marshal` ?

Reply:
Just want to make the JSON text output more like what we used to see? 😜
```
The difference between json.Marshal() and json.MarshalIndent() is that the former does a compact output without indentation, the latter applies (somewhat customizable) indent. 
````
You could see [the example here](https://play.golang.org/p/iMCmfkQARRk) for better declaration . ;).